server/daos/booksLanguages.js

```diff
@@ -0,0 +1,45 @@
+import db from '@database/models';
+import { isEmpty } from 'lodash';
+
+export const insertBooksLanguages = async args => {
+  if (!isEmpty(args.booksLanguages)) {
+    const mapBooksLanguagesArgs = args.booksLanguages.map((item, index) => ({
+      bookId: args.bookId,
+      languageId: item.languageId
+    }));
+
+    const res = await db.booksLanguages.bulkCreate(mapBooksLanguagesArgs);
+
+    return res;
+  }
+};
+
+export const updateBooksLanguagesForBooks = async args => {
+  if (!isEmpty(args.booksLanguages)) {
+    const mapBooksLanguagesArgs = args.booksLanguages.map((item, index) => ({
+      bookId: args.bookId,
+      languageId: item.languageId
+    }));
+
+    await db.booksLanguages.destroy({ where: { bookId: args.bookId } });
+
+    const res = await db.booksLanguages.bulkCreate(mapBooksLanguagesArgs);
+
+    return res;
+  }
+};
+
+export const updateBooksLanguagesForLanguages = async args => {
+  if (!isEmpty(args.booksLanguages)) {
+    const mapBooksLanguagesArgs = args.booksLanguages.map((item, index) => ({
+      bookId: item.bookId,
+      languageId: args.languageId
+    }));
+
+    await db.booksLanguages.destroy({ where: { languageId: args.languageId } });
```

Comment:
hmm this doesn't seem right. Assume that you have languageId = 1 for 100 different books. This will destroy all of them

Reply:
I'll rectify it